src/query/src/sql.rs

```diff
@@ -235,6 +249,100 @@ fn describe_column_semantic_types(
     ))
 }
 
+fn show_create_column_name(table_name: Vec<String>) -> VectorRef {
+    Arc::new(StringVector::from(table_name))
+}
+
+fn show_create_table_column_name(table_info: TableInfoRef) -> VectorRef {
+    let mut create_sql_column: Vec<String> = Vec::new();
+
+    let columns_schemas = table_info.meta.schema.column_schemas();
+    let primary_key_indices = &table_info.meta.primary_key_indices;
+    let table_name = &table_info.name;
+
+    create_sql_column.push(format!("CREATE TABLE {}(", table_name));
+    for (index, cs) in columns_schemas.iter().enumerate() {
+        create_sql_column.push(format!("{} ", cs.name.clone()));
+        create_sql_column.push(format!("{} ", cs.data_type.name()));
+        let default_val = if cs.is_nullable() {
+            "null".to_string()
+        } else if cs.default_constraint().is_none() {
+            "".to_string()
+        } else {
+            format!(
+                "default {}",
+                cs.default_constraint()
+                    .map_or(String::from(""), |dc| dc.to_string())
+            )
+        };
+        create_sql_column.push(default_val);
+        if index != columns_schemas.len() - 1 {
+            create_sql_column.push(", ".to_string());
+        }
+    }
+
+    let mut create_sql_index: Vec<String> = Vec::new();
+    for (index, cs) in columns_schemas.iter().enumerate() {
+        if primary_key_indices.contains(&index) {
+            create_sql_index.push(format!("PRIMARY KEY({})", cs.name));
+            create_sql_index.push(", ".to_string());
+        } else if cs.is_time_index() {
+            create_sql_index.push(format!("TIME INDEX({})", cs.name));
+            create_sql_index.push(", ".to_string());
+        } else {
+        };
+    }
+
+    if !create_sql_index.is_empty() {
+        create_sql_column.push(", ".to_string());
+        create_sql_index.remove(create_sql_index.len() - 1);
+        create_sql_index.push(")".to_string());
+    }
+
+    create_sql_index.push(format!(
+        "ENGINE={} WITH(REGIONS={});",
+        table_info.meta.engine,
+        table_info.meta.region_numbers.len()
+    ));
+
+    let create_sql_column_value = create_sql_column.join("");
```

Comment:
What about creating a [`CreateTable`](https://github.com/GreptimeTeam/greptimedb/blob/c5b0d2431f4095c7a429e84124a594e4fd56df78/src/sql/src/statements/create.rs#L21) statement from the table info and implementing `Display` for it? Then we can use the `to_string()` method to get a `CREATE TABLE` SQL. `sqlparser-rs` library uses similar [approach](https://docs.rs/sqlparser/latest/sqlparser/ast/enum.Statement.html#impl-Display-for-Statement) to render its statement back to SQL.